lib/utils.pm

```diff
@@ -1011,35 +1011,47 @@ be matched (regex) and the answer with string or key to be typed. for example:
         string => "testpasspw\n",
       },]
 
-A "Script done." message comes from the typescript will be printed as a mark
+A "EOS~~~" message followed by return value will be printed as a mark
 for the end of interaction after the command finished running.
+
+If the first argument is undef, only the sencond part will be processed - to
+match output and react. If the second argument is undef, the first part will
+be processed - to run the command without interaction with terminal output.
+This is useful for some situation when you want to do more between inputing
+command and the following interaction, eg. switch TTYs or detach the screen.
+
 =cut
 sub script_run_interactive {
     my ($cmd, $scan, $timeout) = @_;
     my $output;
+    my $err_ret;
     my @words;
+    my $endmark = 'EOS~~~';    # EOS == "End of Script"
     $timeout //= 180;
 
     if ($cmd) {
-        script_run("script -c \'", 0);
-        script_run($cmd,           0);
-
-        # Write to /dev/null since we want not to leave file there
-        script_run("\' /dev/null |& tee /dev/$serialdev", 0);
+        script_run("(script -qe -a /dev/null -c \'", 0);
+        script_run($cmd,                             0);
```

Comment:
I suggest to write in one line like: 
script_run($cmd."; echo $endmark\$?) |& tee /dev/$serialdev", 0);

Reply:
It is intended to do so, so that the wrapped command could be shown in a new line, to make it more clear in the screen output. It is quite similar to what `validate_script_output` do.